azure-pipelines.release-fluentui.yml

```diff
@@ -131,6 +131,15 @@ jobs:
     steps:
       - template: .devops/templates/tools.yml
 
+      - task: CmdLine@2
+        displayName: Checkout branch for pull
+        condition: and(succeeded(), eq(variables.publishOfficial, true))
+        inputs:
+          script: |
+            BRANCH_NAME=`echo $(Build.SourceBranch) | sed "s/refs\/heads\///"`
```

Comment:
I think you should be able to do this with a [`checkout` step](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#checkout) in the job itself, or possibly by modifying the `repositories` config earlier in the file.

Reply:
I didn't find anything like that in azure docs. But when searching for 'why azure checkes out detached head', I read [here](https://github.com/Microsoft/azure-pipelines-agent/issues/1980#issuecomment-440428831) that it is by design, to make sure the pipeline always build the correct commit, even when someone pushed a change when the build is queued.